src/handlers/transfers/handler.js

```diff
@@ -540,9 +540,15 @@ const processFulfilMessage = async (message, functionality, span) => {
           throw fspiopError
         }
       } else {
-        const payeeAccount = await Participant.getAccountByNameAndCurrency(transfer.payeeFsp, transfer.currency, Enum.Accounts.LedgerAccountType.POSITION)
-        await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail, messageKey: payeeAccount.participantCurrencyId.toString(), topicNameOverride, hubName: Config.HUB_NAME })
-        histTimerEnd({ success: true, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
+        const participantCurrencyId = await ProxyCache.deriveCurrencyId(transfer.payeeFsp, transfer.currency)
+        if (participantCurrencyId) {
+          await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail, messageKey: participantCurrencyId.toString(), topicNameOverride, hubName: Config.HUB_NAME })
```

Comment:
Would you not need a `checkSameCreditorDebtorProxy` check here when publishing the messageKey to see if the key needs to be `0`?

Reply:
You are right, I missed it. Its added now. Thanks